src/cloudWatchLogs/registry/logStreamRegistry.ts

```diff
@@ -190,23 +195,32 @@ export async function filterLogEventsFromUriComponents(
     logGroupInfo: CloudWatchLogsGroupInfo,
     parameters: CloudWatchLogsParameters,
     nextToken?: string
-): Promise<CloudWatchLogsResponse> {
+): Promise<CloudWatchLogsResponse | void> {
     const client = new DefaultCloudWatchLogsClient(logGroupInfo.regionName)
 
-    const response = await client.filterLogEvents({
+    const timeout = new Timeout(5000) // How should I set this? This waits up to 5 seconds now
```

Comment:
Should be as long as a user might conceivably want to wait, perhaps minutes, or infinite.

Reply:
So I can simply set this in the code? Should it ideally be user-configurable?